tests/docker/docker_test.go

```diff
@@ -0,0 +1,219 @@
+package lim
+
+import (
+	"flag"
+	"fmt"
+	"github.com/docker/docker/pkg/namesgenerator"
+	"github.com/lf-edge/eden/pkg/defaults"
+	"github.com/lf-edge/eden/pkg/device"
+	"github.com/lf-edge/eden/pkg/expect"
+	"github.com/lf-edge/eden/pkg/projects"
+	"github.com/lf-edge/eden/pkg/utils"
+	"github.com/lf-edge/eve/api/go/info"
+	log "github.com/sirupsen/logrus"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+)
+
+// This test deploys the docker://nginx app into EVE with port forwarding 8028->80
+// wait for the RUNNING state and checks access to HTTP endpoint
+// and removes app from EVE
+
+var (
+	timewait     = flag.Int("timewait", 600, "Timewait for items waiting in seconds")
+	tc           *projects.TestContext
+	externalIP   string
+	appName      string
+	externalPort = 8028
+	appLink      = "docker://nginx"
+	portPublish  = []string{fmt.Sprintf("%d:80", externalPort)}
+)
+
+// TestMain is used to provide setup and teardown for the rest of the
+// tests. As part of setup we make sure that context has a slice of
+// EVE instances that we can operate on. For any action, if the instance
+// is not specified explicitly it is assumed to be the first one in the slice
+func TestMain(m *testing.M) {
+	fmt.Println("Docker app deployment Test")
+
+	tc = projects.NewTestContext()
+
+	projectName := fmt.Sprintf("%s_%s", "TestLogInfoMetric", time.Now())
+
+	tc.InitProject(projectName)
+
+	for _, node := range tc.GetNodeDescriptions() {
+		edgeNode := tc.GetController().GetEdgeNode(node.Name)
+		if edgeNode == nil {
+			edgeNode = tc.NewEdgeNode(tc.WithNodeDescription(node), tc.WithCurrentProject())
+		} else {
+			tc.UpdateEdgeNode(edgeNode, tc.WithCurrentProject(), tc.WithDeviceModel(node.Model))
+		}
+
+		tc.ConfigSync(edgeNode)
+
+		if edgeNode.GetState() == device.NotOnboarded {
+			log.Fatal("Node is not onboarded now")
+		}
+
+		tc.AddNode(edgeNode)
+	}
+
+	tc.StartTrackingState(false)
+
+	res := m.Run()
+
+	os.Exit(res)
+}
+
+//checkAppMsg wait for info of ZInfoApp type with mention of deployed AppName
+func checkAppMsg(appName string) projects.ProcInfoFunc {
```

Comment:
Maybe also library? We need it in VNC case also? Also maybe get it to State? Also the name is misleading, 
Rename for checkAppDeployStarted

Reply:
Renamed. As for me, it is not so interesting for library.